package/ports_daemon/ports_daemon/src/ports.c

```diff
@@ -97,17 +102,21 @@ typedef struct port_config_s {
   pid_t adapter_pid; /* May be cleared by SIGCHLD handler */
   restart_type_t restart;
   bool first_start;
+  const int blacklist_len;
```

Comment:
just realized this isn't strictly necessary but could use a macro (defined in so many ways throughout codebases) like this: `#define ARRAY_SIZE(a) (sizeof(a) / sizeof((a)[0]))`